### DISCUSSION TITLE
How does MOOSE distinguish 2D plane problem?

### DISCUSSION THREAD

**Discussion by Yaode-Yin** [idaholab/moose#26964] [Q&A Modules: Solid mechanics] [answered]
Hi, everyone! I am a beginner for Moose. When I solve one element tension in 2D case, I find that the young's modules evaluated from the stress-strain curve calculated by Moose is different form that as the input material parameter. If the input young's modules is E0 and the Poisson's ratio is v0, the young's modulus calculated E seems to meet E=E0 /(1-v0^2).  
For instance, I use the `ComputeIsotropicElasticityTensor` material, e.g.
```
 [./elasticity_tensor]
    type = ComputeIsotropicElasticityTensor
    youngs_modulus = 210e3
    poissons_ratio = 0.3
  [../]
```
![image](https://github.com/idaholab/moose/assets/146060023/f9200722-34f6-4bd0-aa05-84459b499295)
The input young's modulus is 210e3. But in the one 2D element tension experiment, the slope of stress-train curve is about  230e3. If I set the Poisson's ratio as 0, the slope of stress-train curve is about  210e3 correctly. 
![image](https://github.com/idaholab/moose/assets/146060023/610a80cb-1248-47f0-8a50-e3b2cadb87bf)
I wonder why the Poisson's ratio has an effect on the stress-strain curve of the one element tension in 2D case? The default 2D plane problem in Moose is plane stress or plane strain? And How could I set it?

**Comment by adhithya96** [comment]
Hi Yaode-Yin, 
How did you apply the boundary conditions only at the corner points ?

**Reply by Yaode-Yin** [reply]
Sorry to be late. Yes, the boundary conditions are only applied on the corner points (displacement-control loading on y direction). The numerical results should be in good argument with the exact solution of uniaxial tension.

**Reply by adhithya96** [reply]
Can you share your input file ?  I am also new to MOOSE. Was just curious

**Reply by jiangwen84** [reply, marked as answer]
The default is plane strain.

**Reply by Yaode-Yin** [reply]
If the default is plane strain, how can I solve a plane stress problem in MOOSE? How should I adjust the codes?

**Reply by jiangwen84** [reply]
https://mooseframework.inl.gov/source/kernels/WeakPlaneStress.html

**Reply by Yaode-Yin** [reply]
Is it possible that one uses `WeakPlaneStress` and `PhaseFieldFractureMechanicsOffDiag` kernels together? I want to solve a phase field fracture problem in plane stress case.
